src/solidlsp/language_servers/al_language_server.py

```diff
@@ -0,0 +1,1045 @@
+
+        # Determine platform-specific executable path within the extension
+        # AL provides different binaries for each platform
+        system = platform.system()
+        if system == "Windows":
+            executable_relative_path = os.path.join("extension", "bin", "win32", "Microsoft.Dynamics.Nav.EditorServices.Host.exe")
+        elif system == "Linux":
+            executable_relative_path = os.path.join("extension", "bin", "linux", "Microsoft.Dynamics.Nav.EditorServices.Host")
+        elif system == "Darwin":
+            executable_relative_path = os.path.join("extension", "bin", "darwin", "Microsoft.Dynamics.Nav.EditorServices.Host")
+        else:
+            raise RuntimeError(f"Unsupported platform: {system}")
+
+        executable_path = os.path.join(al_extension_dir, executable_relative_path)
+
+        # Check if the extension is already installed
+        if not os.path.exists(executable_path):
+            logger.log(f"AL Language Server not found at {executable_path}. Downloading from VS Code marketplace...", logging.INFO)
+
+            # AL extension version - using latest stable version
+            AL_VERSION = "latest"
+            url = f"https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-dynamics-smb/vsextensions/al/{AL_VERSION}/vspackage"
+
+            # Try to download the extension
+            if cls._download_al_extension(logger, url, al_extension_dir):
+                logger.log("AL Extension downloaded successfully.", logging.INFO)
+            else:
+                logger.log("Failed to download AL extension, falling back to manual search.", logging.WARNING)
+                # Fall back to the old method of finding the extension
+                return cls._get_language_server_command_fallback(logger)
+
+        # Verify executable exists after download
+        if not os.path.exists(executable_path):
+            logger.log("AL Language Server executable still not found after download. Falling back to manual search.", logging.WARNING)
+            return cls._get_language_server_command_fallback(logger)
+
+        # Make sure executable has proper permissions on Unix-like systems
+        # This is crucial for Linux/macOS where downloaded files may not be executable
+        if system in ["Linux", "Darwin"]:
+            import stat
+
+            st = os.stat(executable_path)
+            os.chmod(executable_path, st.st_mode | stat.S_IEXEC)
+
+        logger.log(f"Using AL Language Server executable: {executable_path}", level=5)
+
+        # The AL Language Server uses stdio for LSP communication by default
+        # Unlike many other language servers, it doesn't require a --stdio flag
+        # On Windows, we need to properly quote the path if it contains spaces to avoid shell issues
+        if system == "Windows" and " " in executable_path:
+            return f'"{executable_path}"'
```

Comment:
Is it harmful to quote for linux and mac?

The check on ` " " in executable_path` is unnecessary, let's always quote on windows. This is even worth a util method in `common.py`